src/test/custom_node_output_allocator_test.cpp

```diff
@@ -0,0 +1,100 @@
+//*****************************************************************************
+// Copyright 2021 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include "../custom_node_output_allocator.hpp"
+
+using namespace ovms;
+
+class NodeLibraryCheckingReleaseCalled : public NodeLibrary {
+    static bool releaseBufferCalled;
+
+public:
+    ~NodeLibraryCheckingReleaseCalled();
+    static int execute(const struct CustomNodeTensor* inputs, int inputsLength, struct CustomNodeTensor** outputs, int* outputsLength, const struct CustomNodeParam* params, int paramsLength);
+    static int releaseBuffer(struct CustomNodeTensor* output);
+    static int releaseTensors(struct CustomNodeTensor* outputs);
+};
+
+NodeLibraryCheckingReleaseCalled::~NodeLibraryCheckingReleaseCalled() {
+    EXPECT_TRUE(releaseBufferCalled);
```

Comment:
Maybe change `releaseBufferCalled` to public and check its value in test? To force blob to be removed you can call `blob->deallocate()`